### PR TITLE
Initialization and tests

### PR DESCRIPTION
**Pull request by stayman**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# Dependency directories
+node_modules
+jspm_packages
+
+# Optional npm cache directory
+.npm
+
+# Optional REPL history
+.node_repl_history
```
Comment (tarikjn) [inline .gitignore]: looks like this file has more than needed

File: lib/AnimateOut.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return (
+      (this.props.showing || this.state.leaving) ?
+        // see: http://stackoverflow.com/a/35102287
+        React.cloneElement(
+          this.props.children,
+          {onAnimationEnd: this.processAnimationEvent, close: this.close, leaving: this.state.leaving})
+        :null
+    )
+  }
+});
```
Comment (tarikjn) [inline lib/AnimateOut.js]: How does this one work without an export?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "test": "./node_modules/.bin/mocha"
+  },
+  "author": "tarikjn",
+  "license": "MIT",
+  "peerDependencies": {
+    "react": "^15.1.0"
+  },
+  "devDependencies": {
+    "babel-core": "^6.10.4",
+    "babel-preset-react": "^6.5.0",
```
Comment (tarikjn) [inline package.json]: Probably doesn't need babel

File: test/animateout.test.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      this.instance.processAnimationEvent.call(this.instance, 'foo', {animationName: 'foo'});
+      expect(this.instance.state.leaving).to.be.false;
+    });
+
+    it('should not set the leaving state to false if the events don\'t match', function() {
+      this.instance.processAnimationEvent.call(this.instance, 'foo', {});
+      expect(this.instance.state.leaving).to.be.true;
+    });
+  });
+});
```
Comment (tarikjn) [inline test/animateout.test.js]: missing end of file newline (other files bellow too)

File: .babelrc [hunk 5]
```diff
@@ -0,0 +1 @@
+{ "presets": ["react"] }
```
Comment (tarikjn) [inline .babelrc]: Looks like it may not be needed.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "animate-out",
+  "version": "0.0.1",
+  "description": "",
+  "main": "lib/index.js",
+  "scripts": {
+    "test": "./node_modules/.bin/mocha"
+  },
+  "author": "tarikjn",
+  "license": "MIT",
```
Comment (tarikjn) [inline package.json]: license is incorrect

File: lib/AnimateOut.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+var React = require('React');
+
+// NOTE: do not use ease out animation if the modal generates a pending action conflicting with animation or
+//       trigger action in a way not guaranteed to preserve modal
+
+var AnimateOut = module.exports = React.createClass({
```
Comment (stayman) [inline lib/AnimateOut.js]: there's `module.exports` up here

File: .babelrc (hunk 5 above)
Comment (stayman) [inline .babelrc]: Just for preservation, we talked offline about babel. We have it in so we can use jsx in our tests

File: lib/AnimateOut.js (hunk 7 above)
Comment (tarikjn) [inline lib/AnimateOut.js]: my bad! nifty syntax

**Pull request closed without merging**